File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Presentation/ppttopptx.rb
module Asposeslidesjava
  module PptToPptx
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate a Presentation object that represents a PPTX file
        pres = Rjb::import('com.aspose.slides.Presentation').new(data_dir + "Presentation1.ppt")

        # Saving the PPTX presentation to PPTX format
        save_format = Rjb::import('com.aspose.slides.SaveFormat')
        pres.save(data_dir + "Aspose.pptx", save_format.Pptx)

        puts "Document has been converted, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/slides/examples/presentations/conversion/PPTtoPPTX.java
package com.aspose.slides.examples.presentations.conversion;

import com.aspose.slides.Presentation;
import com.aspose.slides.SaveFormat;
import com.aspose.slides.examples.RunExamples;





public class PPTtoPPTX
{
    public static void main(String[] args)
    {
        //ExStart:PPTtoPPTX
        // The path to the documents directory.
        String dataDir = RunExamples.getDataDir_Conversion();

        // Instantiate a Presentation object that represents a PPTX file
        Presentation pres = new Presentation(dataDir + "PPTtoPPTX.ppt");

        // Saving the PPTX presentation to PPTX format
        pres.save(dataDir + "PPTtoPPTX_out.pptx", SaveFormat.Pptx);
        //ExEnd:PPTtoPPTX
    }
}
